hawkbit-sdk/hawkbit-sdk-dmf/src/main/java/org/eclipse/hawkbit/sdk/dmf/DmfController.java

```diff
@@ -83,10 +85,24 @@ public void stop() {
     }
 
     public void processUpdate(final EventTopic actionType, final DmfDownloadAndUpdateRequest updateRequest) {
-        updateHandler.getUpdateProcessor(this, actionType, updateRequest);
+        log.info("Processing update for {} actionId : {}", controllerId, updateRequest.getActionId());
+        updateHandler.getUpdateProcessor(this, actionType, updateRequest).run();
+        updateRequest.getSoftwareModules().forEach(sm -> {
+            updateAttributes(sm.getModuleId().toString(), sm.getModuleVersion());
+        });
     }
 
     public void sendFeedback(final UpdateStatus updateStatus) {
+        log.info("Sending {} for controller : {} action : {}", updateStatus.status(), controllerId, currentActionId);
         dmfSender.sendFeedback(tenantId, currentActionId, updateStatus);
     }
+
+    public void sendUpdateAttributes() {
+        log.info("Sending UPDATE_ATTRIBUTES for controller {}", controllerId);
+        dmfSender.updateAttributes(tenantId, controllerId, DmfUpdateMode.MERGE, attributes);
+    }
+
+    public void updateAttributes(final String key, final String value) {
+        this.attributes.put(key, value);
```

Comment:
better rename to setAttrubute because that describes more precisely what the method does.
you should probably add removeAttribute or allow null value that would be interpreted as "remove attribute with key"

Reply:
done. 
also added removeAttribute method.